frontend/src/components/Common/Modal/Modal.tsx

```diff
@@ -0,0 +1,120 @@
+import React, { useRef, useEffect } from 'react'
+import { XMarkIcon } from '@heroicons/react/24/outline'
+import Button from '../Button/Button'
+
+export interface ModalProps {
+  isOpen: boolean
+  onClose: () => void
+  title: string
+  children: React.ReactNode
+  footer?: React.ReactNode
+  size?: 'sm' | 'md' | 'lg' | 'xl'
+  closeOnClickOutside?: boolean
+}
+
+const Modal: React.FC<ModalProps> = ({
+  isOpen,
+  onClose,
+  title,
+  children,
+  footer,
+  size = 'md',
+  closeOnClickOutside = true
+}) => {
+  const modalRef = useRef<HTMLDivElement>(null)
+  const initialFocusRef = useRef<HTMLButtonElement>(null)
+
+  // Size classes mapping
+  const sizeClasses = {
+    sm: 'max-w-md',
+    md: 'max-w-lg',
+    lg: 'max-w-2xl',
+    xl: 'max-w-4xl'
+  }
+
+  // Focus management
+  useEffect(() => {
+    if (isOpen && initialFocusRef.current) {
+      // Focus the close button when modal opens
+      initialFocusRef.current.focus()
+    }
+  }, [isOpen])
+
+  // Handle click outside
+  useEffect(() => {
+    const handleClickOutside = (event: MouseEvent) => {
+      if (closeOnClickOutside && modalRef.current && !modalRef.current.contains(event.target as Node)) {
+        onClose()
+      }
+    }
+
+    // Handle escape key press
+    const handleEscKey = (event: KeyboardEvent) => {
+      if (event.key === 'Escape') {
+        onClose()
+      }
+    }
+
+    if (isOpen) {
+      document.addEventListener('mousedown', handleClickOutside)
+      document.addEventListener('keydown', handleEscKey)
+      // Prevent scrolling on body when modal is open
+      document.body.style.overflow = 'hidden'
```

Comment:
Please handle this using React conditional rendering instead of DOM manipulation